### PR TITLE
Fix(rum-core): Check if first-contentful-paint exists before setting the agent marks

### PR DESCRIPTION
**Pull request by bjorngi**
https://github.com/elastic/apm-agent-rum-js/issues/680

### CODE REVIEW

File: packages/rum-core/src/performance-monitoring/capture-navigation.js [hunk 1]
```diff
@@ -254,24 +254,21 @@ function getNavigationTimingMarks() {
...
-      const { startTime } = fcpEntry[0]
-      fcp = unloadDiff >= 0 ? startTime - unloadDiff : startTime
-    }
+    const timing = PERF.timing
+    /**
+     * To avoid capturing the unload event handler effect
+     * as part of the transaction duration
+     */
+    const unloadDiff = timing.fetchStart - timing.navigationStart
+    const fcp = PERF.getEntriesByType(PAINT).reduce((_fcp, entry) => {
```
Comment (vigneshshanmugam) [inline packages/rum-core/src/performance-monitoring/capture-navigation.js]: can you add a initial value of `null` to the reduce function as entries can be empty array `[]` and we need an initial value for reduce to work. 

```js
reduce(fcp,entry=> {}, null)
```

Comment (bjorngi) [reply packages/rum-core/src/performance-monitoring/capture-navigation.js]: CLA Signed and initialValue added.

**Comment by vigneshshanmugam** [comment]
jenkins run the test please

**Comment by jahtalab** [comment]
@bjorngi , Regarding the CLA issue, it seems that you signed the CLA with a different email compared to the [email set in the commit](https://github.com/elastic/apm-agent-rum-js/commit/e00b0e3be7e3f4e64dd6850461fd3de06319eae4.patch), would you please sign it with the correct email?

**Comment by bjorngi** [comment]
>     1. would you please change the return value to `undefined`  instead of `null`?
Done

>     2. We generally add or improve tests when ever a new bug is discovered, would you please add some tests to cover these fixes? Please have a look at our [Contributing guide](https://github.com/elastic/apm-agent-rum-js/blob/master/CONTRIBUTING.md) and let us know if you need any help.
Not sure how good the test are. Added new mock for missing 'first-contentful-paint' and another test for base case.

CLA signed with correct email for key.

**Pull request merged**